python/lsst/analysis/tools/actions/scalar/scalarActions.py

```diff
@@ -432,5 +435,7 @@ def __call__(self, data: KeyedData, **kwargs) -> Scalar:
         scalarA = self.actionA(data, **kwargs)
         scalarB = self.actionB(data, **kwargs)
         if scalarB == 0:
-            raise ValueError("Denominator is zero!")
-        return scalarA / scalarB
+            _LOG.warning("Denominator is zero! Returning NaN.")
+            return np.NaN
```

Comment:
Hmm.. `np.nan` might be a better choice here because it clearly indicates an invalid answer due to a mathematical operation that cannot be performed.

Reply:
Done!

Reply:
In terms of general mathematics I agree with Erfan, however I think Tim is correct when it comes to floating point division by zero (see https://en.wikipedia.org/wiki/IEEE_754#Exception_handling for example). So I would recommend making `x/0 = np.sign(x)*np.inf`.

Reply:
Ok, here's where it gets more complicated. It looks like the analysis_tools code uses `np.nanmedian` and `np.nanmean` to calculate the median and mean respectively, which will skip NaNs but not infinities. So that means that summing over columns where all of the infinities have the same sign will return `+/- infinity`, but if any of them happen to have different signs then the result will be NaN.  This seems like suboptimal behavior, so I wonder if Erfan's suggestion will produce more expected behavior, even if not technically correct.

Reply:
True, those are valid considerations.

Reply:
Maybe we should discuss on Slack. Ignoring infinities in your mean seems like a problem whereas ignoring NaN seems correct. Making all your infinities disappear seems wrong mathematically and should instead be telling you something about your metrics.

Reply:
Maybe @erykoff has an opinion.

Reply:
I really could go either way. I completely agree that ignoring infinities can be problematic, but given that `np.inf - np.inf = np.nan`, in most image processing cases summing up quantities divided by zero will result in NaN anyway. So if we're really worried about infinities we should probably check for them explicitly before summing them. I suppose for `nanmedian` we're technically fine because the infinities will be placed at the beginning or end of the sorted sequence depending on the sign, but that just seems like another reason that we should check for infinities explicitly before doing these calculations.